### PR TITLE
[fix]: [CDS-73981]: Proper error message when invalid tags are provided for env/infra filters

### PR DESCRIPTION
**Pull request by lovish1999**
## Describe your changes
We will throw an Invalid request exception along with proper error message in case the filter tags have value as empty or string (i.e. not a Map).

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51297)

### CODE REVIEW

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/environment/helper/FilterTagsUtils.java [hunk 1]
```diff
@@ -42,6 +42,11 @@ public static boolean areAnyTagFiltersMatching(List<NGTag> entityTags, List<NGTa
   public static boolean areTagsFilterMatching(List<NGTag> entityTags, TagsFilter tagsFilter) {
     String matchType = tagsFilter.getMatchType().getValue();
 
+    Object tagsValue = tagsFilter.getTags().getValue();
+    if (!(tagsValue instanceof Map<?, ?>) ) {
+      throw new InvalidRequestException(
+          "Filter Tag expression used for an environment/infrastructure filter is not supported. Supported expressions for filter tags should resolve to a map of strings.");
```
Comment (ivan-83) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/environment/helper/FilterTagsUtils.java:48]: The error msg could be,  Invalid filter tags value found, %s. Filter tags should be key-value pairs. ,
              input.getClass()

or...

**Review by ivan-83 — DISMISSED** [review]

**Review by ivan-83 — DISMISSED** [review]

**Comment by lovish1999** [comment]
trigger uts

**Comment by lovish1999** [comment]
trigger uts

**Comment by lovish1999** [comment]
trigger uts

**Review by ivan-83 — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r2, 1 of 1 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51297#-:-Nauo_eJ-UoPVzGogFdG:bnfp4nl)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Review by tchaurasiya — APPROVED** [review]

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/environment/helper/FilterTagsUtils.java [hunk 2]
```diff
@@ -41,7 +46,12 @@ public static boolean areAnyTagFiltersMatching(List<NGTag> entityTags, List<NGTa
 
   public static boolean areTagsFilterMatching(List<NGTag> entityTags, TagsFilter tagsFilter) {
     String matchType = tagsFilter.getMatchType().getValue();
-
+    Object tagsValue = tagsFilter.getTags().getValue();
+    if (!(tagsValue instanceof Map<?, ?>) ) {
+      throw new InvalidRequestException(String.format(
+          "Invalid filter tags value found [%s]. Filter tags should be non-empty key-value pairs of string values.",
```
Comment (tchaurasiya) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/environment/helper/FilterTagsUtils.java:52]: if tags are expression we should print exception in place of null

**Review by tchaurasiya — CHANGES_REQUESTED** [review]

**Review by tchaurasiya — APPROVED** [review]

**Comment by lovish1999** [comment]
trigger review

**Pull request merged**